Core/include/Acts/AmbiguityResolution/GreedyAmbiguityResolution.ipp

```diff
@@ -0,0 +1,72 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/AmbiguityResolution/GreedyAmbiguityResolution.hpp"
+
+#include <unordered_map>
+
+namespace Acts {
+
+template <typename track_container_t, typename traj_t,
+          template <typename> class holder_t, typename source_link_hash_t,
+          typename source_link_equality_t>
+void GreedyAmbiguityResolution::computeInitialState(
+    const TrackContainer<track_container_t, traj_t, holder_t>& tracks,
+    State& state, source_link_hash_t&& sourceLinkHash,
+    source_link_equality_t&& sourceLinkEquality) const {
+  auto measurementIndexMap =
+      std::unordered_map<SourceLink, std::size_t, source_link_hash_t,
+                         source_link_equality_t>(0, sourceLinkHash,
+                                                 sourceLinkEquality);
+
+  for (const auto& track : tracks) {
+    if (track.nMeasurements() < m_cfg.nMeasurementsMin) {
+      continue;
+    }
+    std::vector<std::size_t> measurements;
+    tracks.trackStateContainer().visitBackwards(
```

Comment:
Just fyi, I plan to add convenience to track that allows you to iterate over the measurements only without having to filter yourself.

Reply:
Also: you don't have to change it, but you can do

```cpp
for (auto ts : track.trackStates()) {}
```

So you don't necessarily have to use the tip index manually.

Would be nice to use so I can see if the interface works well.